backend/src/packages/surveys/survey.service.ts

```diff
@@ -26,6 +29,23 @@ class SurveyService implements Service {
   public async create(
     payload: SurveyRequestDto,
   ): Promise<SurveyGetAllItemResponseDto> {
+    const user = await userService.findById(payload.userId);
+
+    if (!user) {
+      throw new UsersError({
+        status: HTTPCode.NOT_FOUND,
+        message: ExceptionMessage.USER_NOT_FOUND,
+      });
+    }
+
+    if (user.isSurveyCompleted) {
+      const item = (await this.surveyRepository.update(
+        payload,
+      )) as SurveyEntity;
```

Comment:
after the change above, you won't need this
```suggestion
      ));
```